### PR TITLE
React 15 / Jest

### PR DESCRIPTION
**Pull request by xdissent**
I read thru @jonnor's [comment](https://github.com/flowhub/the-graph/issues/360#issuecomment-347663223) and wanted to get the ball rolling on tests that would make a react upgrade feasible. This PR does the minimum required to allow use in strictly-cjs environments (webpack) and upgrade to react-15 with no deprecation warnings. I copied the existing tests from `spec` and began filling them out for jest. If this is the wrong approach, please let me know. Otherwise I'm going to continue adding tests, focussing on maximizing coverage. I did *not* replace the mocha runner with jest yet, so CI and `npm test` still run the old tests for now.

### CODE REVIEW

File: the-graph/the-graph-app.js [hunk 1]
```diff
@@ -437,10 +441,10 @@ module.exports.register = function (context) {
       }
 
       // Wheel to zoom
-      if (domNode.onwheel!==undefined) {
+      if ('onwheel' in domNode) {
         // Chrome and Firefox
         domNode.addEventListener("wheel", this.onWheel);
-      } else if (domNode.onmousewheel!==undefined) {
+      } else if ('onmousewheel' in domNode) {
```
Comment (xdissent) [inline the-graph/the-graph-app.js]: Aside from deprecation fixes, these 2 lines are the only code changes so far.

**Comment by jonnor** [comment]
Hi @xdissent, thanks for the PR! Some feedback based on quick look:

The React deprecation fixes look good.

How to run the Jest tests? I am OK with replacing Mocha with Jest (not much there anyways) but then we need to get it up and running on Travis CI (and delete old tests). What would be the steps needed to do that?
The tests do not have to run on multiple browsers, and can/should probably not use a service Saucelabs (it makes local and PRs painful).

**Comment by jonnor** [comment]
@xdissent  I tried running the tests that were added now, but I get a lot of failures. I did `grunt build` and then `jest`, after a clean `npm install`.
Am I doing something wrong? Complete test runner output below

```
[jon@jon-portable the-graph]$ ./node_modules/.bin/jest
 FAIL  __tests__/editing.js
  ● Editor navigation › dragging on background › should pan graph view

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:47:15)
          at <anonymous>

  ● Editor navigation › dragging on background › should pan graph view

    TypeError: Cannot read property 'state' of undefined
      
      at Object.it (__tests__/editing.js:57:22)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor navigation › dragging on background › should pan graph view

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:51:27)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor navigation › mouse scrolling up › should zoom in

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:47:15)
          at <anonymous>

  ● Editor navigation › mouse scrolling up › should zoom in

    TypeError: Cannot read property 'state' of undefined
      
      at Object.it (__tests__/editing.js:70:22)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor navigation › mouse scrolling up › should zoom in

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:51:27)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor navigation › mouse scrolling down › should zoom out

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:47:15)
          at <anonymous>

  ● Editor navigation › mouse scrolling down › should zoom out

    TypeError: Cannot read property 'state' of undefined
      
      at Object.it (__tests__/editing.js:82:22)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor navigation › mouse scrolling down › should zoom out

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:51:27)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor navigation › multitouch pinch › should zoom in/out

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:47:15)
          at <anonymous>

  ● Editor navigation › multitouch pinch › should zoom in/out

    TypeError: Cannot read property 'state' of undefined
      
      at Object.it (__tests__/editing.js:93:22)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor navigation › multitouch pinch › should zoom in/out

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:51:27)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › dragging on node › should move the node

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:133:15)
          at <anonymous>

  ● Editor › dragging on node › should move the node

    TypeError: Cannot read property 'props' of undefined
      
      at Object.it (__tests__/editing.js:152:22)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › dragging on node › should move the node

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:146:27)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › dragging from node port › should start making edge

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:133:15)
          at <anonymous>

  ● Editor › dragging from node port › should start making edge

    TypeError: Cannot read property 'getElementsByClassName' of undefined
      
      at Object.it (__tests__/editing.js:172:18)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › dragging from node port › should start making edge

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:146:27)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › dropping started edge on port › should connect the edge

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:133:15)
          at <anonymous>

  ● Editor › dropping started edge on port › should connect the edge

    TypeError: Cannot read property 'getElementsByClassName' of undefined
      
      at Object.it (__tests__/editing.js:185:29)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › dropping started edge on port › should connect the edge

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:146:27)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › dropping started edge outside › should not connect the edge

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:133:15)
          at <anonymous>

  ● Editor › dropping started edge outside › should not connect the edge

    TypeError: Cannot read property 'getElementsByClassName' of undefined
      
      at Object.it (__tests__/editing.js:206:7)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › dropping started edge outside › should not connect the edge

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:146:27)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › clicking unselected node › should add node to selection

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:133:15)
          at <anonymous>

  ● Editor › clicking unselected node › should add node to selection

    TypeError: Cannot read property 'getElementsByClassName' of undefined
      
      at Object.it (__tests__/editing.js:225:20)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › clicking unselected node › should add node to selection

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:146:27)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › clicking selected node › should remove node from selection

    TypeError: Cannot read property 'clearRect' of null
      
      at Object.renderCanvas (the-graph/the-graph-app.js:541:9)
      at Object.componentDidUpdate (the-graph/the-graph-app.js:534:12)
      at Object.componentDidMount (the-graph/the-graph-app.js:472:12)
      at Object.chainedFunction [as componentDidMount] (node_modules/create-react-class/factory.js:615:11)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:262:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:261:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:74:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:78:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:207:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:154:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:124:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:141:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:60:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:95:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:317:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:399:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:420:23)
      at Object.render (node_modules/enzyme-adapter-react-15/build/ReactFifteenAdapter.js:165:50)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:98:16)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.beforeEach (__tests__/editing.js:133:15)
          at <anonymous>

  ● Editor › clicking selected node › should remove node from selection

    TypeError: Cannot read property 'getElementsByClassName' of undefined
      
      at Object.it (__tests__/editing.js:234:20)
          at new Promise (<anonymous>)
          at <anonymous>

  ● Editor › clicking selected node › should remove node from selection

    TypeError: Cannot read property 'unmount' of undefined
      
      at Object.afterEach (__tests__/editing.js:146:27)
          at new Promise (<anonymous>)
          at <anonymous>

 PASS  __tests__/basics.js

Test Suites: 1 failed, 3 skipped, 1 passed, 2 of 5 total
Tests:       10 failed, 51 skipped, 3 passed, 64 total
Snapshots:   0 total
Time:        7.011s
Ran all test suites.
```

**Comment by jonnor** [comment]
I see that Jest tests against JsDom in node.js, and doesn't really have a good story for testing in browsers (see https://github.com/facebook/jest/issues/139 and https://github.com/facebook/jest/issues/848). I really don't like testing something different from the execution environment, but I guess coverage is effectively improved by these new test, so...

**Comment by jonnor** [comment]
Tests pass on Travis CI, and passed locally when I installed `canvas` - so was probably a problem with `canvas-prebuilt` on my local machine.

**Pull request merged**